packages/shared/src/tms/projection.tile.matrix.set.ts

```diff
@@ -64,6 +64,20 @@ export class ProjectionTileMatrixSet {
         return (epsgCode && CodeMap.get(epsgCode)) ?? null;
     }
 
+    /**
+     * Project the points in a MultiPolygon array to the `targetProjection`. Modifies `multipoly`.
+     * Does not support holes in polygons.
+     */
+    projectMultipolygon(multipoly: number[][][][], targetProjection: ProjectionTileMatrixSet): void {
```

Comment:
return a new polygon instead of destroying the current one

Reply:
Yeah, I knew you would say that. It just means I have to write more code because I have to copy one that are in same projection because otherwise I break the contract; I hate wasting CPU cycles also.

Reply:
Hmm, could change it so if projection source == target throw? we shouldnt be projecting to the same thing?

in terms of lines, its pretty silimar with 

```
const output = [];
      for (const poly of multipoly) {
            output.push(poly[0].map((p) => fromWsg84(toWsg84(p))));
        }
return output;
```

and you are probably wasting more cpu cycles by using a `.map()` inside of a loop and creating a  function inside the loop too.

Reply:
Now returns same multipoly if same projection otherwise projects to new projection